### PR TITLE
[FLINK-23626] [Runtime / Metrics] Migrate all TaskMG instantiations to factory method

### PR DESCRIPTION
**Pull request by hililiwei**
## What is the purpose of the change

Modify all existing usages of the TaskMG constructor to use runtime Apis, for consistency and to make constructor changes easier.

## Verifying this change

This change is already covered by existing tests, such as *(please describe tests)*.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no

### CODE REVIEW

**Review by zentol — CHANGES_REQUESTED** [review]
This does not change the code as intended.

We do not want a _new_ factory method; instead the `TaskMetricGroup` should be created via `TaskManagerJobMetricgroup#addTask`.

**Comment by hililiwei** [comment]
@flinkbot run azure

File: flink-runtime/src/test/java/org/apache/flink/runtime/operators/chaining/ChainedOperatorsMetricTest.java [hunk 1]
```diff
@@ -68,22 +70,26 @@
...
-                                        new ExecutionAttemptID(),
-                                        "task",
-                                        0,
-                                        0))
+                                new TaskManagerJobMetricGroup(
+                                                NoOpMetricRegistry.INSTANCE,
+                                                UnregisteredMetricGroups
+                                                        .createUnregisteredTaskManagerMetricGroup(),
+                                                new JobID(0, 0),
+                                                "UnregisteredJob")
```
Comment (zentol) [inline flink-runtime/src/test/java/org/apache/flink/runtime/operators/chaining/ChainedOperatorsMetricTest.java:86]:
```suggestion
                                UnregisteredMetricGroups
                                                .createUnregisteredTaskManagerJobMetricGroup()
```

**Review by zentol — CHANGES_REQUESTED** [review]
Mostly good, had 1 small comment.

File: flink-runtime/src/test/java/org/apache/flink/runtime/operators/chaining/ChainedOperatorsMetricTest.java (hunk 1 above)
Comment (zentol) [reply flink-runtime/src/test/java/org/apache/flink/runtime/operators/chaining/ChainedOperatorsMetricTest.java:86]: Or actually could we just use a `#createUnregisteredTaskMetricGroup`?

Comment (zentol) [reply flink-runtime/src/test/java/org/apache/flink/runtime/operators/chaining/ChainedOperatorsMetricTest.java:86]: urgh the test relies on the same OperatorMG being returned...

File: flink-runtime/src/test/java/org/apache/flink/runtime/operators/chaining/ChainedOperatorsMetricTest.java [hunk 2]
```diff
@@ -68,22 +70,26 @@
...
+                                        .addTask(
+                                                new JobVertexID(),
+                                                new ExecutionAttemptID(),
+                                                "task",
+                                                0,
+                                                0))
```
Comment (zentol) [inline flink-runtime/src/test/java/org/apache/flink/runtime/operators/chaining/ChainedOperatorsMetricTest.java:92]:
```suggestion
                                MetricUtils.createTaskManagerMetricGroup(
                                                NoOpMetricRegistry.INSTANCE,
                                                "host",
                                                ResourceID.generate())
                                        .addTaskForJob(
                                                new JobID(),
                                                "jobname",
                                                new JobVertexID(),
                                                new ExecutionAttemptID(),
                                                "task",
                                                0,
                                                0))
```
lets do it like this.

**Comment by hililiwei** [comment]
@flinkbot run azure

File: flink-runtime/src/test/java/org/apache/flink/runtime/operators/chaining/ChainedOperatorsMetricTest.java (hunk 2 above)
Comment (hililiwei) [reply flink-runtime/src/test/java/org/apache/flink/runtime/operators/chaining/ChainedOperatorsMetricTest.java]: Thanks for your direction

**Comment by hililiwei** [comment]
@flinkbot run azure

**Pull request merged**